core/src/main/java/com/yahoo/ycsb/Client.java

```diff
@@ -309,57 +309,76 @@ public void run()
     //NOTE: Switching to using nanoTime and parkNanos for time management here such that the measurements
     // and the client thread have the same view on time.
 
-    //spread the thread operations out so they don't all hit the DB at the same time
-    // GH issue 4 - throws exception if _target>1 because random.nextInt argument must be >0
-    // and the sleep() doesn't make sense for granularities < 1 ms anyway
-    if ((_targetOpsPerMs > 0) && (_targetOpsPerMs <= 1.0))
-    {
-      long randomMinorDelay = Utils.random().nextInt((int) _targetOpsTickNs);
-      sleepUntil(System.nanoTime() + randomMinorDelay);
-    }
-    try
-    {
-      if (_dotransactions)
-      {
-        long startTimeNanos = System.nanoTime();
-
-        while (((_opcount == 0) || (_opsdone < _opcount)) && !_workload.isStopRequested())
-        {
-
-          if (!_workload.doTransaction(_db,_workloadstate))
-          {
-            break;
-          }
-
-          _opsdone++;
-
-          throttleNanos(startTimeNanos);
-        }
-      }
-      else
-      {
-        long startTimeNanos = System.nanoTime();
-
-        while (((_opcount == 0) || (_opsdone < _opcount)) && !_workload.isStopRequested())
-        {
-
-          if (!_workload.doInsert(_db,_workloadstate))
-          {
-            break;
-          }
-
-          _opsdone++;
-
-          throttleNanos(startTimeNanos);
-        }
-      }
-    }
-    catch (Exception e)
-    {
-      e.printStackTrace();
-      e.printStackTrace(System.out);
-      System.exit(0);
-    }
+		//spread the thread operations out so they don't all hit the DB at the same time
+		try
+		{
+		   //GH issue 4 - throws exception if _target>1 because random.nextInt argument must be >0
+		   //and the sleep() doesn't make sense for granularities < 1 ms anyway
+		   if ( (_targetOpsPerMs>0) && (_targetOpsPerMs<=1.0) ) 
+		   {
+		      sleep(Utils.random().nextInt((int)(1.0/_targetOpsTickNs)));
```

Comment:
this looks unrelated to the change mentioned in the PR description, could you leave it out?